src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/AbstractExpressionHandler.java

```diff
@@ -500,6 +501,10 @@ protected void checkModifiers() {
              modifier != null;
              modifier = modifier.getNextSibling()) {
             if (isOnStartOfLine(modifier)
+                 && (!TokenUtil.isOfType(modifier,
+                        TokenTypes.ANNOTATION, TokenTypes.STRICTFP, TokenTypes.LITERAL_STATIC)
+                    || modifier.getPreviousSibling() == null
+                    || !isPrecededByAnyModifier(modifier))
```

Comment:
@romani @nrmancuso 

I am continuing on the discussion at https://github.com/checkstyle/checkstyle/issues/14716#issuecomment-2028461416 , I don't see how skipping specific tokens will serve a benefit. I don't think we should be reviewing this PR until we have some agreement here.

Including @nrmancuso 's original comments, the input file in this PR needs to be vastly expanded.

We need a duplicate input file with all indentation at 0 showing the violations produced by bad code. Regression was done on good formatted code, so it will not provide less value for this check in those terms. We should also do force strict as it will tell us exactly what indentation should be and when it is absolutely correct and not just meeting the min requirement.

We also need inputs where these tokens being skipped are completely alone on a line and are the first AST in the class def. All the current examples start with public, abstract, final which are not any of the tokens listed here.

Reply:
>There doesn't seem to be any clear reason why we are targeting these tokens

Yep, my sentiments exactly.